homeassistant/components/fish_audio/config_flow.py

```diff
@@ -0,0 +1,390 @@
+        api_key = self._get_entry().data[CONF_API_KEY]
+        try:
+            _, self.session = await _validate_api_key(self.hass, api_key)
+        except InvalidAuthError:
+            return self.async_abort(reason="invalid_auth")
+        except CannotConnectError:
+            return self.async_abort(reason="cannot_connect")
+        except UnexpectedError:
+            return self.async_abort(reason="unknown")
+
+        if user_input is not None:
+            self.config_data.update(user_input)
+            return await self.async_step_model()
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=get_filter_schema(self.config_data),
+            errors={},
+        )
+
+    async def async_step_model(
+        self, user_input: SubentryModelUserInput | None = None
+    ) -> SubentryFlowResult:
+        """Handle the model selection step."""
+        errors: dict[str, str] = {}
+
+        if not self.models:
+            try:
+                self.models = await self._async_get_models(
+                    self_only=self.config_data.get(CONF_SELF_ONLY, False),
+                    language=self.config_data.get(CONF_LANGUAGE),
+                    sort_by=self.config_data.get(CONF_SORT_BY, "score"),
+                )
+            except CannotGetModelsError:
+                self.models = []
+
+            if not self.models:
+                errors["base"] = "no_models_found"
+
+        if user_input is not None:
+            if (voice_id := user_input.get(CONF_VOICE_ID)) and (
+                backend := user_input.get(CONF_BACKEND)
+            ):
+                self.config_data.update(user_input)
+                if self._is_new:
+                    voice_name = next(
+                        (m["label"] for m in self.models if m["value"] == voice_id),
+                        "Fish Audio TTS",
+                    )
+
+                    return await self.async_step_name(default=voice_name)
+
+                unique_id = f"{voice_id}-{backend}"
+
+                return self.async_update_and_abort(
+                    self._get_entry(),
+                    self._get_reconfigure_subentry(),
+                    data=self.config_data,
+                    unique_id=unique_id,
+                )
+
+        return self.async_show_form(
+            step_id="model",
+            data_schema=get_model_selection_schema(self.config_data, self.models),
+            errors=errors,
+        )
+
+    async def async_step_name(
+        self,
+        user_input: SubentryNameUserInput | None = None,
+        default: str | None = None,
+        unique_id: str | None = None,
+    ) -> SubentryFlowResult:
+        """Handle the name selection step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            self.config_data.update(user_input)
+            unique_id = (
+                f"{self.config_data[CONF_VOICE_ID]}-{self.config_data[CONF_BACKEND]}"
+            )
```

Comment:
One thing to keep in mind, is there a reason for someone to pick the same voice and backend? If so, I am not so sure about the unique id. If not, we should abort if it's already used